### PR TITLE
Implement pagination for events

### PR DESCRIPTION
**Pull request by alexkocharyan01**

### CODE REVIEW

File: src/pages/events/index.tsx [hunk 1]
```diff
@@ -1,45 +1,25 @@
...
-]
+const PAGE_BOTTOM = 600
 
 const Events = () => {
-  const [showModeId, setShowModeId] = useState<number | null>()
+  const [page, setPage] = useState<number>(1)
+  const { eventsData, pageSize, isEventsLoading, totalItems } = useAppSelector(
+    eventsSelector.events
+  )
+  const pageCout = totalItems / pageSize
```
Comment (gor017) [inline src/pages/events/index.tsx:20]: Fix typo

File: src/redux/slices/events/middleware.ts [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+
+const { setEventsLoading, setError, setEventsData, setPageSize, setTotalItems } = slice.actions
+
+const events = (currentPage: number) => async (dispatch: AppDispatch) => {
+  try {
+    dispatch(setEventsLoading(true))
+
+    const response = await API.events.getEvents(currentPage)
+
+    console.log(response, 'response')
```
Comment (gor017) [inline src/redux/slices/events/middleware.ts:15]: remove console

File: src/redux/slices/events/middleware.ts [hunk 3]
```diff
@@ -0,0 +1,29 @@
...
+    dispatch(setEventsData(response.data.data))
+    dispatch(setPageSize(response.data.pageSize))
+    dispatch(setEventsLoading(false))
+    dispatch(setTotalItems(response.data.totalItems))
```
Comment (gor017) [inline src/redux/slices/events/middleware.ts:19]: let's add a config to take data from response.data, for rejecting write response.data in each case

**Review by gor017 — CHANGES_REQUESTED** [review]

File: src/pages/events/index.tsx (hunk 1 above)
Comment (alexkocharyan01) [reply src/pages/events/index.tsx]: done

File: src/redux/slices/events/middleware.ts (hunk 2 above)
Comment (alexkocharyan01) [reply src/redux/slices/events/middleware.ts]: done

File: src/manager/events/authManager.ts [hunk 4]
```diff
@@ -0,0 +1,15 @@
...
+import { IAxiosResponsePaginated } from '@axios/axiosTypes'
+
+const axiosInstance = Axios()
+const authManager = {
+  axiosInstance,
+  getEvents(currentPage: number) {
+    return axiosInstance.get<IEventsDataType, IAxiosResponsePaginated<IEventsDataType[]>>(
+      `core/v1/event?page=${currentPage}`
+    )
+  },
```
Comment (gor017) [inline src/manager/events/authManager.ts:12]: pass object as params

File: src/types/reduxTypes/eventsStateTypes.ts [hunk 5]
```diff
@@ -0,0 +1,20 @@
+export interface IEventsProps {
+  events: IEvents
+}
+
+export interface IEvents {
+  isEventsLoading: boolean
+  error: string
```
Comment (gor017) [inline src/types/reduxTypes/eventsStateTypes.ts:7]: Can error be null?

File: src/types/reduxTypes/eventsStateTypes.ts [hunk 6]
```diff
@@ -0,0 +1,20 @@
...
+  eventsData: IEventsDataType[]
+  pageSize: number
+  totalItems: number
+}
+
+export interface IEventsDataType {
+  id: string
```
Comment (gor017) [inline src/types/reduxTypes/eventsStateTypes.ts:14]: Should be uuid

File: src/redux/slices/events/reducers.ts [hunk 7]
```diff
@@ -0,0 +1,29 @@
...
+  },
+
+  setEventsLoading(state, action: IAction<boolean>) {
+    state.events.isEventsLoading = action.payload
+  },
+
+  setEventsData(state, action: IAction<IEventsDataType[]>) {
+    state.events.eventsData = [...state.events.eventsData, ...action.payload]
+  },
+
```
Comment (gor017) [inline src/redux/slices/events/reducers.ts:19]: Move merge logic too middleware

**Review by gor017 — CHANGES_REQUESTED** [review]

File: src/types/reduxTypes/eventsStateTypes.ts [hunk 8]
```diff
@@ -0,0 +1,20 @@
...
+
+export interface IEvents {
+  isEventsLoading: boolean
+  error: string
+  eventsData: IEventsDataType[]
+  pageSize: number
+  totalItems: number
+}
+
+export interface IEventsDataType {
```
[the remark this answers is not in the document]
Comment (alexkocharyan01) [reply src/types/reduxTypes/eventsStateTypes.ts:13]: done

File: src/types/reduxTypes/eventsStateTypes.ts (hunk 5 above)
Comment (alexkocharyan01) [reply src/types/reduxTypes/eventsStateTypes.ts:7]: done

File: src/types/reduxTypes/eventsStateTypes.ts (hunk 6 above)
Comment (alexkocharyan01) [reply src/types/reduxTypes/eventsStateTypes.ts:14]: done

File: src/redux/slices/events/reducers.ts (hunk 7 above)
Comment (alexkocharyan01) [reply src/redux/slices/events/reducers.ts:19]: done

File: src/types/reduxTypes/eventsStateTypes.ts [hunk 9]
```diff
@@ -0,0 +1,24 @@
+export interface IEventsProps {
+  events: IEvents
+}
+
+export interface IEvents {
+  isEventsLoading: boolean
+  error: string | null
+  eventsData: IEventsDataProps[]
```
Comment (gor017) [inline src/types/reduxTypes/eventsStateTypes.ts:8]: If it will we events list change name to eventsList

**Review by gor017 — CHANGES_REQUESTED** [review]

File: src/manager/events/authManager.ts [hunk 10]
```diff
@@ -0,0 +1,20 @@
...
+    return axiosInstance.get<IEventsDataProps, IAxiosResponsePaginated<IEventsDataProps[]>>(
+      `${baseURL}`,
+      {
+        params,
+      }
+    )
+  },
+}
+
+export default eventsManager
```
[the remark this answers is not in the document]
Comment (alexkocharyan01) [reply src/manager/events/authManager.ts:20]: done

File: src/types/reduxTypes/eventsStateTypes.ts (hunk 9 above)
Comment (alexkocharyan01) [reply src/types/reduxTypes/eventsStateTypes.ts:8]: done

File: src/redux/slices/events/middleware.ts [hunk 11]
```diff
@@ -0,0 +1,36 @@
+import API from '@axios/API'
+import { IError } from '@axios/authentication/authManagerTypes'
+import store, { AppDispatch } from '@redux/store'
+
+import slice from './slice'
+
+const { setEventsLoading, setError, setEventsData, setPageSize, setTotalItems } = slice.actions
+
+const events = (page: number) => async (dispatch: AppDispatch) => {
```
[the remark this answers is not in the document]
Comment (alexkocharyan01) [reply src/redux/slices/events/middleware.ts:9]: done

File: src/pages/events/index.tsx [hunk 12]
```diff
@@ -1,45 +1,24 @@
...
-    image: '/image1.jpg',
-    isButtonVisible: false,
-  },
-]
+const PAGE_BOTTOM = 600
 
 const Events = () => {
-  const [showModeId, setShowModeId] = useState<number | null>()
+  const [page, setPage] = useState<number>(1)
+  const { eventsData, pageSize, isEventsLoading, totalItems } = useAppSelector(
```
[the remark this answers is not in the document]
Comment (alexkocharyan01) [reply src/pages/events/index.tsx]: done

File: src/redux/slices/events/reducers.ts [hunk 13]
```diff
@@ -0,0 +1,29 @@
...
+    state.events.error = action.payload
+  },
+
+  setEventsLoading(state, action: IAction<boolean>) {
+    state.events.isEventsLoading = action.payload
+  },
+
+  setEventsData(state, action: IAction<IEventsDataProps[]>) {
+    state.events.eventsData = action.payload
+  },
```
[the remark this answers is not in the document]
Comment (alexkocharyan01) [reply src/redux/slices/events/reducers.ts]: done

**Review by gor017 — APPROVED** [review]

**Pull request merged**